src/main/java/Duke.java

```diff
@@ -1,10 +1,88 @@
+                int taskNumber = Integer.parseInt(line.substring(divider + 1));
+                list[taskNumber - 1].setDone();
+                System.out.println("Nice! I've marked this task as done:");
+                System.out.println("  " + list[taskNumber - 1]);
+            //If user inputted deadline, then add the task under deadline cat
+            } else if (line.contains("deadline")) {
+                int dividerAction = line.indexOf(" ");
+                int dividerDate = line.indexOf("/");
+                list[noOfTask] = new Deadline(line.substring((dividerAction + 1),
+                        (dividerDate - 1)), line.substring(dividerDate + 4));
+                noOfTask++;
+                System.out.println("Got it. I've added this task:");
+                System.out.println("  " + list[noOfTask - 1]);
+                if ((noOfTask - 1) == 0) {
+                    System.out.println("Now you have " + noOfTask + " task in the list.");
+                } else {
+                    System.out.println("Now you have " + noOfTask + " tasks in the list.");
+                }
+            //If user inputted to do, then add the task under to do cat
+            } else if (line.contains("todo")) {
+                int divider = line.indexOf(" ");
+                list[noOfTask] = new Todo(line.substring(divider + 1));
+                noOfTask++;
+                System.out.println("Got it. I've added this task:");
+                System.out.println("  " + list[noOfTask - 1]);
+                if ((noOfTask - 1) == 0) {
+                    System.out.println("Now you have " + noOfTask + " task in the list.");
+                } else {
+                    System.out.println("Now you have " + noOfTask + " tasks in the list.");
+                }
+            //If user inputted event, then add task under event cat
+            } else if (line.contains("event")) {
+                int dividerAction = line.indexOf(" ");
+                int dividerDate = line.indexOf("/");
+                list[noOfTask] = new Event(line.substring((dividerAction + 1),
+                        (dividerDate - 1)), line.substring(dividerDate + 4));
+                noOfTask ++;
+                System.out.println("Got it. I've added this task:");
+                System.out.println("  " + list[noOfTask - 1]);
+                if ((noOfTask - 1) == 0) {
+                    System.out.println("Now you have " + noOfTask + " task in the list.");
+                } else {
+                    System.out.println("Now you have " + noOfTask + " tasks in the list.");
+                }
+            //else print added statement and add the item to list
+            } else {
+                System.out.println("added: " + line);
+                list[noOfTask] = new Task(line);
+                noOfTask ++;
+            }
+            line = in.nextLine();
+        }
+        System.out.println("Bye. Hope to see you again soon!");
     }
```

Comment:
I feel that breaking some parts into smaller functions will improve readability

Reply:
yup! I will implement the slap algorithm